src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Logging/IRazorLoggerFactoryExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.CodeAnalysis.Razor.Logging;
+
+internal static class IRazorLoggerFactoryExtensions
+{
+    public static ILogger CreateLogger<T>(this IRazorLoggerFactory factory)
+    {
+        return factory.CreateLogger(TrimTypeName(typeof(T).Name));
+    }
+
+    private static string TrimTypeName(string name)
+    {
+        string trimmedName;
+        _ = TryTrim(name, "Microsoft.VisualStudio.", out trimmedName) ||
+            TryTrim(name, "Microsoft.AspNetCore.Razor.", out trimmedName);
+
+        return trimmedName;
+
+        static bool TryTrim(string name, string prefix, out string trimmedName)
+        {
+            if (prefix.StartsWith(name, StringComparison.Ordinal))
```

Comment:
Should this be `name.StartsWith(prefix`?

Reply:
and below

Reply:
lol wait what. I swear I've seen this working.

Reply:
oh lol, I changed from `typeof(T).FullName` to `typeof(T).Name` accidentally. Maybe Name is enough though actually?

Reply:
Maybe. What happens for collisions?

Reply:
Yeah, probably better to have slightly more noise. We can always adjust later. Have fixed.